### ISSUE TITLE
Unexpected result of `model_eval` (inconsistent with keras implementation) 

### ISSUE THREAD

**Issue by martinmatak** [tensorflow/cleverhans#559]
What I noticed is that `cleverhans.utils_tf.model_eval` function gives unexpected result, 
i.e. result is not consistent with result of `keras.models.Model.evaluate`. 

Here is my code so that result can be reproduced: 

```
import tensorflow as tf
from keras.models import load_model
from cleverhans.utils_keras import KerasModelWrapper
from cleverhans.utils_tf import model_eval
from cleverhans.utils_mnist import data_mnist
from keras.optimizers import SGD
from keras.layers import Dense, Flatten
from keras.layers import Conv2D, MaxPooling2D
from keras.models import Sequential

num_classes = 10
batch_size = 128
nb_epochs = 2 # to reproduce results faster
img_size = 28
nb_channels = 1


def main():
    # Create TF session
    sess = tf.Session()

    # load data
    x_train, y_train, x_test, y_test = data_mnist()

    # load model
    # model_path = 'same result if we load a model instead of training it'
    # model = load_model(model_path)
    # print("Model loaded.")

    # train a model
    model = get_keras_model(num_classes, (img_size, img_size, nb_channels))
    sgd = SGD(lr=0.1, momentum=0.9, decay=0.5)
    model.compile(loss='categorical_crossentropy', optimizer=sgd, metrics=['accuracy'])
    model.fit(x_train, y_train,
              batch_size=batch_size,
              epochs=nb_epochs,
              validation_data=(x_test, y_test),
              shuffle=True)


    # Score trained model via keras impl of evaluate method
    scores = model.evaluate(x_test, y_test, verbose=1)
    print('Test loss:', scores['loss'])
    print('Test accuracy:', scores['acc'])

    # initialize placeholders
    x = tf.placeholder(tf.float32, shape=(None, img_size, img_size, nb_channels))
    y = tf.placeholder(tf.float32, shape=(None, num_classes))

    # wrap it
    model = KerasModelWrapper(model)

    # initialize session
    init_op = tf.global_variables_initializer()
    sess.run(init_op)

    # Score trained model (again) vie keras implementation of evaluate
    scores = model.model.evaluate(x_test, y_test, verbose=1)
    print('Test loss (again):', scores[0])
    print('Test accuracy (again):', scores[1])

    # evaluate wrapper via model_eval
    predictions = model.get_logits(x)
    accuracy = model_eval(sess, x, y, predictions, x_test, y_test, args={'batch_size': batch_size})
    print('Test accuracy (cleverhans impl): ' + str(accuracy))

def get_keras_model(num_classes, input_shape):
    model = Sequential()
    model.add(Conv2D(64, kernel_size=(3, 3), strides=(1, 1),
                     activation='relu',
                     input_shape=input_shape))
    model.add(Conv2D(64, kernel_size=(3, 3), strides=(1, 1),
                     activation='relu',
                     input_shape=input_shape))
    model.add(MaxPooling2D(pool_size=(2, 2), strides=(2, 2)))
    model.add(Conv2D(128, (3, 3), activation='relu'))
    model.add(Conv2D(128, (3, 3), activation='relu'))
    model.add(MaxPooling2D(pool_size=(2, 2)))
    model.add(Flatten())
    model.add(Dense(256, activation='relu'))
    model.add(Dense(256, activation='relu'))
    model.add(Dense(num_classes, activation='softmax'))
    return model


if __name__ == '__main__':
    main()
```

and output (unimportant messages are omitted) it produces is the following:

```
Test loss: 0.12419448612220585
Test accuracy: 0.9633

Test loss (again): 0.12419448612220585
Test accuracy (again): 0.9633

Test accuracy (cleverhans impl): 0.1708
```

How is it the case that those accuracies are not the same?

**Comment by goodfeli** [comment]
It's likely to be a bug in KerasModelWrapper. Have you tried running the keras Sequential model on one batch and checking that its output is the same as the output of KerasModelWrapper?

Is there any chance that Keras has made its own session in the backend? CleverHans might be reading the values of the model's Variables out of one Session while Keras reads them out of another.

**Comment by goodfeli** [comment]
What version of keras are you running? I'm using keras 2.2.2 and when I run your code I get an exception when trying to access `scores['loss']`. For me, `scores` is a list.
